### PR TITLE
Ltd 979 license status query

### PR DESCRIPTION
**Pull request by dgnball**
Add an endpoint to lite-hmrc that takes a license id and returns back the mail status.

### CODE REVIEW

File: mail/views.py [hunk 1]
```diff
@@ -111,3 +110,10 @@ class SetAllToReplySent(APIView):
     def get(self, _):
         Mail.objects.all().update(status=ReceptionStatusEnum.REPLY_SENT)
         return HttpResponse(status=HTTP_200_OK)
+
+
+class License(APIView):
+    def get(self, request):
+        license_id = request.GET.get("id", "")
+        mail = list(LicenceData.objects.filter(licence_ids__contains=license_id))[0].mail
+        return JsonResponse({"status": mail.status})
```
Comment (alixedi) [inline mail/views.py:119]: I would recommend writing a model serializer if we don't already have one and [using DRF to do the work for you](https://www.django-rest-framework.org/api-guide/generic-views/#examples).

File: mail/tests/test_end_to_end.py [hunk 2]
```diff
@@ -40,3 +42,6 @@ def test_send_email_to_hmrc_e2e(self):
 8\end\licence\7
 9\fileTrailer\1"""
         assert body == expected_mail_body  # nosec
+        encoded_reference_code = quote("GBSIEL/2020/0000001/P", safe="")
+        response = self.client.get(f"/mail/license/?id={encoded_reference_code}")
+        assert "reply_pending" == response.json()["status"]  # nosec
```
Comment (alixedi) [inline mail/tests/test_end_to_end.py:47]: Generally, we use the `reverse` method that comes with Django to work out the URL. You don't even need to call the `quote` if you use it.

Also, a small one but I think generally the standard way to write assertions is -

```
assert response.json()["status"] == "reply_pending"
```

(It is a check whether reality is the same as expectations and so it reads better in that order)

File: mail/views.py [hunk 3]
```diff
@@ -111,3 +110,10 @@ class SetAllToReplySent(APIView):
     def get(self, _):
         Mail.objects.all().update(status=ReceptionStatusEnum.REPLY_SENT)
         return HttpResponse(status=HTTP_200_OK)
+
+
+class License(APIView):
+    def get(self, request):
+        license_id = request.GET.get("id", "")
+        mail = list(LicenceData.objects.filter(licence_ids__contains=license_id))[0].mail
```
Comment (eadpearce) [inline mail/views.py:118]: if it's only expected to ever return 1 object you can use .get() instead

File: mail/views.py [hunk 4]
```diff
@@ -111,3 +110,10 @@ class SetAllToReplySent(APIView):
     def get(self, _):
         Mail.objects.all().update(status=ReceptionStatusEnum.REPLY_SENT)
         return HttpResponse(status=HTTP_200_OK)
+
+
+class License(APIView):
```
Comment (eadpearce) [inline mail/views.py:115]: if we're just returning the status of the mail a better name would be LicenseMailStatus

**Review by alixedi — COMMENTED** [review]
We like to organize our commits into small atomic changes that read well in `git blame` e.g. this work can be squashed into a single commit - "Add endpoint to returns the list of licenses."

File: mail/views.py (hunk 1 above)
Comment (dgnball) [reply mail/views.py:119]: Don't we get diminishing returns from using a serializer if we are only pulling out a single field?

File: mail/views.py (hunk 3 above)
Comment (dgnball) [reply mail/views.py:118]: Good spot! Will do.

File: mail/views.py (hunk 4 above)
Comment (dgnball) [reply mail/views.py:115]: I see your point and I did conisder something like this but the URL is already /mail/license and I don't know if we want more mail-license related info up the pipe in the future so I'm going to leave the name unless people feel very strongly about it

File: mail/tests/test_end_to_end.py (hunk 2 above)
Comment (dgnball) [reply mail/tests/test_end_to_end.py:47]: Happy to swap the assert round. Not that happy about using reverse. I prefer to see the URL in it's natural state as it provides free documentation of how to use the endpoint. I don't really see what benefit reverse gives us.

**Comment by dgnball** [comment]
> We like to organize our commits into small atomic changes that read well in `git blame` e.g. this work can be squashed into a single commit - "Add endpoint to returns licenses."

ok, will do

File: mail/tests/test_end_to_end.py (hunk 2 above)
Comment (saruniitr) [reply mail/tests/test_end_to_end.py]: `reverse` is better. If we change the path eg `/mail/licence/` to `/main/something-else/etc/etc` then we don't have to change here, it is a best practice.

File: mail/views.py (hunk 1 above)
Comment (saruniitr) [reply mail/views.py]: We don't fully know what details we want to export using this endpoint.
I agree `serializer` is better then the same can be used in tests.

File: mail/tests/test_end_to_end.py (hunk 2 above)
Comment (dgnball) [reply mail/tests/test_end_to_end.py]: I don't think you can use reverse with query strings. I tried and it didn't work for me.

Comment (dgnball) [reply mail/tests/test_end_to_end.py]: See https://gist.github.com/benbacardi/227f924ec1d9bedd242b

File: mail/views.py (hunk 1 above)
Comment (dgnball) [reply mail/views.py]: Ok, done.

File: mail/tests/test_end_to_end.py (hunk 2 above)
Comment (alixedi) [reply mail/tests/test_end_to_end.py]: Good catch! I reckon we should still use `reverse` but use f-strings to pop query string params into the URL.

Comment (dgnball) [reply mail/tests/test_end_to_end.py]: Ok, done!

**Review by saruniitr — APPROVED** [review]
just spelling change required for License.
Every where it is `Licence` but here it is `License`.

Once that is correct it can be merged.

**Pull request merged**